instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/webview/WebViewTrackingE2ETests.kt

```diff
@@ -0,0 +1,159 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.nightly.webview
+
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.espresso.web.sugar.Web.onWebView
+import androidx.test.espresso.web.webdriver.DriverAtoms.findElement
+import androidx.test.espresso.web.webdriver.DriverAtoms.webClick
+import androidx.test.espresso.web.webdriver.DriverAtoms.webScrollIntoView
+import androidx.test.espresso.web.webdriver.Locator
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.core.configuration.Configuration
+import com.datadog.android.nightly.activities.WebViewTrackingActivity
+import com.datadog.android.nightly.activities.WebViewTrackingBridgeHostsActivity
+import com.datadog.android.nightly.rules.NightlyTestRule
+import com.datadog.android.nightly.utils.initializeSdk
+import com.datadog.android.nightly.utils.measureSdkInitialize
+import org.junit.After
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+internal class WebViewTrackingE2ETests {
+
+    @get:Rule
+    val nightlyTestRule = NightlyTestRule()
+
+    @After
+    fun tearDown() {
+        waitForWebViewEvents()
+    }
+
+    // region Tests
+
+    /**
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#constructor()
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#fun getAllowedWebViewHosts(): String
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#fun send(String)
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun setWebViewTrackingHosts(List<String>): Builder
+     */
+    @Test
+    fun web_view_page_view_tracking() {
+        initSdk()
+        launch(WebViewTrackingActivity::class.java)
+
+        // just to make sure the WebView loaded
+        onWebView().perform(webScrollIntoView())
+    }
+
+    /**
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#constructor(List<String>)
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#fun getAllowedWebViewHosts(): String
+     * apiMethodSignature: com.datadog.android.webview.DatadogEventBridge#fun send(String)
+     */
+    @Test
+    fun web_view_page_view_tracking_allowed_hosts_at_bridge_level() {
```

Comment:
It'd be nice to also add a test `web_view_page_view_tracking_host_not_allowed`, where you don't allow the host (neither in config and in the bridge constructor), and make sure the data is never sent to DD